scene/resources/packed_scene.cpp

```diff
@@ -1671,6 +1671,7 @@ bool PackedScene::can_instance() const {
 }
 
 Node *PackedScene::instance(GenEditState p_edit_state) const {
+	MutexLock lck(mtx);
```

Comment:
I understand this is the easy way out, but I honestly don't understand why this function is not reentrant.
I think focus should be put in figuring out why it's not (specially when not using gen edit state)

Reply:
Multiple threads building instances from the same resource doesn't feel very reentrant to me, especially since the PackedScene and SceneState classes store and change so much state during instancing. Given instancing covers a couple hundred lines of code I figured this the sane approach. I don't mind if you don't accept the pull request. I simply saw no way of making it reentrant without refactoring that entire 300 LoC SceneState::instance function. Maybe push this to 3.1 since I don't have time to do the refactoring before 3.0 release?

Reply:
Could it be that some of the member variables are mutable?

Reply:
The node_path_cache is modified during instancing in SceneState. I did not know the mutable keyword violates method constness. https://www.geeksforgeeks.org/c-mutable-keyword/

If multiple threads simultaneously modify the cache, could it crash?

Reply:
My educated guess would be that the cache can be put in an invalid state when multiple threads read and write it simultaneously. Also is alloca threadsafe?